src/peft/tuners/lora.py

```diff
@@ -1131,8 +1152,58 @@ def __init__(
             self.update_layer(adapter_name, r, lora_alpha, lora_dropout, init_lora_weights)
             self.active_adapter = adapter_name
 
+        def merge(self):
+            if self.active_adapter not in self.lora_A.keys():
+                return
+            if self.merged:
+                warnings.warn("Already merged. Nothing to do.")
+                return
+            if self.r[self.active_adapter] > 0:
+                lora_data = self.get_delta_weight(self.active_adapter)
```

Comment:
See my comment above, I think that we need to warn users that merging quantized models can lead to getting different generations due to rounding errors